File: official/docs/java/rates/RetrieveRate.java
package rates;

import com.easypost.EasyPost;
import com.easypost.exception.EasyPostException;
import com.easypost.model.Rate;

public class RetrieveRate {
    public static void main(String[] args) throws EasyPostException {
        EasyPost.apiKey = System.getenv("EASYPOST_API_KEY");

        Rate rate = Rate.retrieve("rate_...");

        System.out.println(rate);
    }
}

File: official/docs/csharp/rates/retrieve-rate.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Xunit;
using Xunit.Sdk;
using Newtonsoft.Json;
using EasyPost;

namespace EasyPostExamples;

public class Examples
{
    [Fact]
    public async Task RetrieveRate()
    {
        string apiKey = Environment.GetEnvironmentVariable("EASYPOST_API_KEY")!;

        EasyPost.ClientManager.SetCurrent(apiKey);

        Rate rate = await Rate.Retrieve("rate...");

        new TestOutputHelper().WriteLine(JsonConvert.SerializeObject(rate, Formatting.Indented));
    }
}
